conbench/templates/app.html

```diff
@@ -179,6 +179,69 @@ <h4 class="modal-title"><span class="glyphicon glyphicon-trash submenu"></span>&
     });
 }
 
+/*
+INFO:
+    Adds free text search ability for specific columns, based on selected column indexes.
+    Based on the indexes of the columns, input boxes are created. The search is triggered on
+    key press.
+INPUTS:
+ data_table_id: string
+ column_indexes: array
+ */
+function free_text_column_search(data_table_id, column_indexes) {
+    $(document).ready(function() {
+        var table = $(data_table_id).DataTable();
+        table.columns(column_indexes).every(function() {
+            var that = this;
+            var title = this.header().textContent;
+            $(`<input class="custom-input-text custom-search-box" type="text" placeholder="🔍 ${title}"
+                onClick="if(event.stopPropagation!==undefined){event.preventDefault(); event.stopPropagation();}event.cancelBubble=true;"/>`)
+                .appendTo(this.header())
+                .on('keyup change clear', function() { //column search/sorting triggered on keypress
+                            that
+                            .search(this.value)
+                            .draw();
+            });
+        });
+    })
+}
+
+/*
+INFO:
+    Function to check unique value count and set the required search type.
+        - Unique values <10 & >1 [specific_column_search]: drop-down based search is implemented
+        - Unique values >10 [free_text_column_search]: free text search is implemented
+        - If there is no data OR only single data point: a message is displayed ("only one/no value found")
+INPUT:
+    data_table_id: string
+ */
+function column_search_implementation(data_table_id) {
+    $(document).ready(function() {
+        var table = $(data_table_id).DataTable();
+        var unique_count = [];
+        table.columns().every( function () {
+            var current = this;
+            //array of unique count for data points within each column
+            unique_count.push(current.data().unique().count())
+        });
+        unique_count.forEach(function (item, index) {
+            if (item < 10 && item > 1) {
+                dropdown_column_search(data_table_id, index)
+            }
+            else if (item > 10){
+                free_text_column_search(data_table_id, index)
+            }
+            else if (item == 1) {
+                $(`<p class="message-header"> 👀 only one/no value found</p>`).appendTo(
```

Comment:
I don't think we need this placeholder; it would probably look cleaner to just do nothing. :)

<img width="1161" alt="image" src="https://user-images.githubusercontent.com/16600275/215522467-a459f142-2a40-47a9-a18d-46fedfa67ad7.png">

Reply:
Yeah, agreed. I'm fine with no placeholder if that's simplest (someone can sort and re-sort if they do believe that there should be more | should be a search bar). But if we do put one it would be nice if it was small enough that een on this column it didn't cause overflowing on lines (maybe a simple `-` or something?)